bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/handler/HaperoBridgeHandler.java

```diff
@@ -0,0 +1,615 @@
+            } catch (InterruptedException e) {
+                logger.debug("Exception in FileWatcherTask: {}", e.getMessage());
+                faultOccurred = true;
+                break;
+            }
+
+            /*
+             * watchService.poll return null, timeout occurred. Switch to OFFLINE
+             * and wait for the next update
+             */
+            if (key == null) {
+                logger.warn("Upload.hld not updated withing timeout.");
+
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.update.timeout");
+                continue;
+            } else {
+                /* A key was returned, check if a modify event for the Upload.hld is present */
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    if (event.kind().equals(StandardWatchEventKinds.ENTRY_MODIFY)) {
+                        if (event.context().toString().equals(HaperoBindingConstants.DATA_FILENAME)) {
+                            fileUpdated = true;
+                            logger.trace("File update detected");
+                        }
+                    }
+                }
+
+                /* If Upload.hld was modified, trigger a State update */
+                if (fileUpdated) {
+                    updateThings();
+                }
+
+                /* Either way, restart the Watcher */
+                if (!key.reset()) {
+                    logger.debug("Key reset failed");
+                    faultOccurred = true;
+                    break;
+                }
+            }
+        }
+
+        /* We broke out of the while(true) loop, so a fault occurred. Try to clean up and set the thing to offline */
+        try {
+            if (watchService != null) {
+                watchService.close();
+            }
+        } catch (IOException e) {
+            logger.debug("Exception in FileWatcherTask: {}", e.getMessage());
+        }
+
+        if (faultOccurred) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/offline.monitoring.failed");
+        }
+    };
+
+    /**
+     * Constructor
+     *
+     * @param bridge
+     */
+    public HaperoBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof HaperoThingHandler) {
+            devicesToNotify.remove(childHandler);
+            logger.trace("Bridge child disposed");
+        } else {
+            logger.debug("childHandler not of type HaperoThingHandler");
+        }
+        super.childHandlerDisposed(childHandler, childThing);
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof HaperoThingHandler haperoThingHandler) {
+            devicesToNotify.add(haperoThingHandler);
```

Comment:
Is the specialized type needed here, but not in `childHandlerDisposed`?

Reply:
I cant say. This is right out of the generated template for a binding. I think i have added the specialized type because of an earlier review comment here.

Reply:
But its not throwing any errors and seems to work fine, if thats what youre asking